unstructured/partition/epub.py

```diff
@@ -29,13 +37,49 @@ def partition_epub(
         If True, the output will include page breaks if the filetype supports it
     metadata_last_modified
         The last modified date for the document.
-
     """
-    return convert_and_partition_html(
-        source_format="epub",
-        filename=filename,
-        file=file,
-        include_page_breaks=include_page_breaks,
-        metadata_filename=metadata_filename,
-        metadata_last_modified=metadata_last_modified,
-    )
+    exactly_one(filename=filename, file=file)
+    if filename is None:
+        filename = ""
+
+    if file is not None:
+        with tempfile.NamedTemporaryFile(delete=False) as tmp:
+            tmp.write(file.read())
+            filename = tmp.name
+        last_modification_date = get_last_modified_date_from_file(file)
+    else:
+        last_modification_date = get_last_modified_date(filename)
+
+    book = epub.read_epub(filename)
+    html_items = [item for item in book.items if isinstance(item, epub.EpubHtml)]
+    toc_href_and_title = []
+    elements = []
+
+    for item in book.toc:
+        # Some toc items may be tuple of multiple items, but all have the same href
+        if isinstance(item, tuple):
+            toc_href_and_title.append((item[0].href.split("#")[0], item[0].title))
+        else:
+            toc_href_and_title.append((item.href.split("#")[0], item.title))
+
+    for item in html_items:
+        # not all html_items show up in the toc,
+        # so some elements will still have `None` for metadata.section
+        item_title = None
+        item_content = item.get_content().decode()
```

Comment:
We should accept an `encoding` kwarg in `partition_epub` and decode using the appropriate encoding. @christinestraub has done a lot of work on encoding already and may be able to help with that.

https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/file_utils/encoding.py

Reply:
I believe I took this snippet `item.get_content().decode()` from Camgreenburg's function, but looking at it more closely, it doesn't seem the `.decode()` part is actually needed (or used--I stepped through `pdb` and didn't see it called and can't find a definition for it through VSC). 
I did find the `.get_content()` definition in ebiooklib, but it doesn't accept an argument for encoding:
```    
def get_content(self, default=None):
        """
        Returns content for this document as HTML string. Content will be of type 'str' (Python 2)
        or 'bytes' (Python 3).

        :Args:
          - default: Default value for the content if it is not defined.

        :Returns:
          Returns content of this document.
        """
```

Would you want to create a fork so we can update this to accept an encoding argument?

Reply:
Nope, I was mistaken about the `.decode()`. It is being used--I must have missed it when stepping through `pdb` (though I'm not sure why it's not being mapped to a definition in VSC)

Reply:
When I use `detect_file_encoding(filename)` on the test epub, I'm getting 'iso-8859-1'. I am wondering if it's necessary for me to use that, though, since it seems to slow down the test significantly. Could it be hard-coded?

This is what I have added locally for encoding, but was hoping to meet with or hear from Christine to take a look at it:
```
        if encoding:
            item_content = item.get_content().decode(encoding)
        else:
            item_content = item.get_content().decode(formatted_encoding)
```

Reply:
@MthwRobinson @Coniferish I noticed that the `EpubHtml.get_content` method always returns `utf-8` encoded content (type `bytes`). https://github.com/aerkalov/ebooklib/blob/master/ebooklib/epub.py#L437
```
tree_str = etree.tostring(tree, pretty_print=True, encoding='utf-8', xml_declaration=True)
```
IMO, we don't have to use the `detect_file_encoding` as long as we use the `EpubHtml.get_content`.

Reply:
Sounds good. I suspected that, but am glad the second opinion!

Reply:
Sounds good - thanks for checking that out!